src/pages/Governance/ProposalList/CreateProposalModal/UploadOrManualProposal/index.tsx

```diff
@@ -0,0 +1,67 @@
+/** @jsxImportSource @emotion/react */
+import { Box } from '@mui/material';
+import { NoticeInfo, SecondaryButton } from 'components';
+import React, { useRef } from 'react';
+import { useTranslation } from 'translation';
+
+import { useStyles } from './styles';
+import TEST_IDS from '../testIds';
+
+interface UploadOrManualProposalProps {
+  onClickUploadFile: (file: File | null) => void;
+  onClickCreateManually: () => void;
+}
+
+const UploadOrManualProposal: React.FC<UploadOrManualProposalProps> = ({
+  onClickCreateManually,
+  onClickUploadFile,
+}) => {
+  const inputFile = useRef<HTMLInputElement | null>(null);
+  const { t, Trans } = useTranslation();
+  const styles = useStyles();
+
+  return (
+    <div>
+      <Box css={styles.options}>
+        <input
+          type="file"
+          accept=".json"
+          id="proposalFile"
+          ref={inputFile}
+          css={styles.fileInput}
+          data-testid={TEST_IDS.fileInput}
+          onChange={e => {
+            e.stopPropagation();
+            e.preventDefault();
+            const file = e.target.files ? e.target.files[0] : null;
+            onClickUploadFile(file);
+          }}
```

Comment:
I'm generally not a fan of having inline functions in the return statement of a component, I feel it makes it harder to read. How about moving this to a function outside the return statement?